### PR TITLE
Added helper method to set OS Deployment Plan in ServerProfile

### LINKED ISSUE #212: Unable to create a Server Profile with Deployment Plan settings
### Scenario/Intent
I'm not able to create any Server Profile using an existing Deployment Plan.
There is no way to get the URI of a Deployment Plan, unless copying it from the GUI.

The GET /rest/os-deployment-plans/{id}, from the OneView API 300, returns two URIs in the response data:

```json
"nativePlanUri": "/rest/oedeploymentplan/c85a5181-8c4c-421c-a018-5b5e69082bce"
```
and

```json
"uri": "/rest/os-deployment-plans/18d42e6d-f64e-422b-affd-7185603ab9c4"
```

The `nativePlanUri` is the same URI from ImageStreamer, and it doesn't work for Server Profile creation.
I must pass in the `uri`, which is the canonical URI.

Follow an example of the Deployment Setting from an existing Server Profile:
```json
    "osDeploymentSettings": {
        "osDeploymentPlanUri": "/rest/os-deployment-plans/18d42e6d-f64e-422b-affd-7185603ab9c4",
        "osVolumeUri": "/rest/deployed-targets/c9bc7b3c-e916-425f-9dad-6c304e1140e1",
        "osCustomAttributes": [
            {
                "name": "hostname",
                "value": "newhostname"
            }
        ]
    }
```

Thus, the endpoints from the OneView Rest API 300 are really useful and must be implemented as soon as possible.

### Environment Details
 - **OneView SDK Version:** any
 - **OneView Appliance Version:** 3.0 / Synergy
 - **OneView Client API Version:** 300
 - **Ruby Version:** any
 - **Platform:** Linux/Windows

### Steps to Reproduce
Call `OneviewSDK::API300::Synergy::OSDeploymentPlan.find_by` (or anything similar, not sure about the class name)

### Expected Result
Return a list of OS Deployment Plans.

### Actual Result
This resource is not implemented yet.

### PR DESCRIPTION
**Pull request by ricardogpsf**
### Description
Added helper method to set OS Deployment Plan in ServerProfile: method "set_os_deployment_plan"

### Issues Resolved
Fixes #212 

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass (`$ rake test`).
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [ ] Changes are documented in the CHANGELOG.

### CODE REVIEW

**Comment by ricardogpsf** [comment]
The Changelog already was updated with description of this issue.

File: spec/integration/resource/api300/synergy/server_profile/create_spec.rb [hunk 1]
```diff
@@ -1,11 +1,12 @@
 # Requirements:
 # => 2 Server Hardware Types (any name)
-# => Enclosure Group 'EnclosureGroup_1'
+# => Enclosure Group 'EnclosureGroup_1' (configured for work with Deployment Network)
```
Comment (tmiotto) [inline spec/integration/resource/api300/synergy/server_profile/create_spec.rb]: (configured with HPE Image Streamer deployment network)

File: lib/oneview-sdk/resource/api300/synergy/server_profile.rb [hunk 2]
```diff
@@ -72,6 +72,18 @@ def self.get_sas_logical_jbod_attachment(client, name)
           results = get_sas_logical_jbod_attachments(client)
           results.find { |attachment| attachment['name'] == name }
         end
+
+        # Sets the OS deployment settings applicable when deployment is invoked through server profile
+        # @param [OneviewSDK::API300::Synergy::OSDeploymentPlan] os_deployment_plan the OSDeploymentPlan resource with valid URI
+        # @param [Array<Hash<String, String>>] custom_attributes The custom attributes to be configured on the OS deployment plan
+        # @option custom_attributes [String] :name name of attribute
```
Comment (tmiotto) [inline lib/oneview-sdk/resource/api300/synergy/server_profile.rb]: `@option` tags work for arrays?

**Comment by ricardogpsf** [comment]
@tmiotto said 

> But I guess we still need to propagate these changes to Server Profile Template! (Consider implementing these changes in a module to avoid code duplication)

But I cannot see the option to set os deployment settings in the api documentation of ServerProfileTemplate. This does work for ServerProfileTemplate?

**Comment by tmiotto** [comment]
I had a vague rememberance of some template with OS deployment plan, maybe it was just my imagination...

File: lib/oneview-sdk/resource/api300/synergy/server_profile.rb (hunk 2 above)
Comment (ricardogpsf) [inline lib/oneview-sdk/resource/api300/synergy/server_profile.rb]: the result is this: 
```
Parameters:
os_deployment_plan (OneviewSDK::API300::Synergy::OSDeploymentPlan) — the OSDeploymentPlan resource with valid URI
custom_attributes (Array<Hash<String, String>>) (defaults to: []) — The custom attributes to be configured on the OS deployment plan
Options Hash (custom_attributes):
:name (String) — name of attribute
:value (String) — value of attribute
```

Comment (tmiotto) [inline lib/oneview-sdk/resource/api300/synergy/server_profile.rb]: Try renaming `custom_attributes` to  `custom_attributes.entries` this way it won't confuse who's reading the documentation.

**Comment by fgbulsoni** [comment]
I'd suggest fixing the changelog and getting it ready for release on this PR as well.

Under `Suggested release: v4.1.1`, I see:
`
New Resources:
(none)
`
But there's also
`
New Resources:
OS Deployment Plan
`

We should bump up suggested release to 4.2 since it adds a new resource and Move that new resource to the upper part of the changelog.

File: lib/oneview-sdk/resource/api300/synergy/server_profile.rb (hunk 2 above)
Comment (ricardogpsf) [inline lib/oneview-sdk/resource/api300/synergy/server_profile.rb]: I guess the result is bad:
image

Comment (ricardogpsf) [inline lib/oneview-sdk/resource/api300/synergy/server_profile.rb]: I suggest this:
image

Comment (tmiotto) [inline lib/oneview-sdk/resource/api300/synergy/server_profile.rb]: How about using something like this:
```ruby
        # @param [Array(Hash<String, String>)] custom_attributes The custom attributes to be configured on the OS deployment plan.
        #   The internal hashes may contain:
        #   - 'name' [String] name of the attribute
        #   - 'value' [String] value of the attribute
```

Comment (ricardogpsf) [inline lib/oneview-sdk/resource/api300/synergy/server_profile.rb]: yeah!

**Comment by fgbulsoni** [comment]
> Prepare for 4.2.0 release

Changes to changelog, readme and version look good to me. Is anything missing in this PR?

**Comment by tmiotto** [comment]
:taco::shipit:

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,8 +1,10 @@
 # Unreleased Changes
-## Suggested release: v4.1.1
+(none)
```
Comment (tmiotto) [inline CHANGELOG.md]: Remove the (none) and unreleased changes, we don't want it on our official gem release!

Comment (ricardogpsf) [inline CHANGELOG.md]: hahaha...Sorry!

Comment (tmiotto) [inline CHANGELOG.md]: No problem!

**Pull request merged**